### PR TITLE
feat: add eddsa MPCv2 SMC utils

### PR DESCRIPTION
**Pull request by kisslove-dewangan**
TICKET: WCI-242

### CODE REVIEW

File: modules/sdk-core/src/bitgo/utils/tss/eddsa/SMC/utils.ts [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+      userMsg1: ovc1.ovcMsg1,
+      backupMsg1: ovc2.ovcMsg1,
+    });
+
+    const response = {
+      state: EddsaKeyCreationMPCv2StateEnum.WaitingForOVC1Round2Data,
+      tssVersion: payload.tssVersion,
+      walletType: payload.walletType,
+      coin: payload.coin,
+      ovc: payload.ovc,
```
Comment (Marzooqa) [inline modules/sdk-core/src/bitgo/utils/tss/eddsa/SMC/utils.ts:65]: **Bug: unsafe `as` cast hiding a type definition gap.**

`EddsaMPCv2KeyGenRound1Response` is typed as `{ sessionId, bitgoMsg1 }` only — `walletGpgPubKeySigs` is not declared on it. The cast `(result as EddsaMPCv2KeyGenRound1Response & { walletGpgPubKeySigs: string })` works around this gap.

For ECDSA, `MPCv2KeyGenRound1Response` properly declares `walletGpgPubKeySigs: NonEmptyString` and is accessed directly without any cast.

Two scenarios:
1. **If the EdDSA `/mpc/keygen/round1` API _does_ return `walletGpgPubKeySigs`** → `EddsaMPCv2KeyGenRound1Response` in `@bitgo/public-types` needs to be updated to include the field, and this cast should be removed.
2. **If the EdDSA API does _not_ return this field** → `walletGpgPubKeySigs` will be `undefined` at runtime. If `EddsaBitgoToOVC1Round1Response.platform.walletGpgPubKeySigs` is a required `NonEmptyString` (as it is for the ECDSA equivalent), `decodeOrElse` will throw `"error(s) parsing response"` on every call, making `keyGenRound1BySender` permanently broken.

Either fix the type in `@bitgo/public-types` or remove `walletGpgPubKeySigs` from the platform construction if EdDSA doesn't use it.

File: modules/sdk-core/src/bitgo/utils/tss/eddsa/SMC/utils.ts [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    const ovc1 = payload.ovc[OVCIndexEnum.ONE];
+    const ovc2 = payload.ovc[OVCIndexEnum.TWO];
+    const sessionId = payload.platform.sessionId;
+    const result = await this.MPCv2Utils.sendKeyGenerationRound2BySender(senderFn, {
+      sessionId,
+      userMsg2: ovc1.ovcMsg2,
+      backupMsg2: ovc2.ovcMsg2,
+    });
+
+    assert.equal(sessionId, result.sessionId, 'Round 1 and round 2 session IDs do not match');
```
Comment (Marzooqa) [inline modules/sdk-core/src/bitgo/utils/tss/eddsa/SMC/utils.ts:98]: **Pattern inconsistency with ECDSA for `sessionId` in round 2.**

ECDSA `keyGenRound2BySender` explicitly overwrites the session ID from the API result:
```typescript
platform: {
  ...payload.platform,
  sessionId: result.sessionId,  // updated from result
  bitgoCommitment2: ...,
}
```

This PR asserts equality (`assert.equal(sessionId, result.sessionId, ...)`) and then keeps the original via `...payload.platform` — the response never includes `result.sessionId` directly.

These two choices are consistent with _each other_ (assert equality, then propagate original), but it's unclear if the assertion is correct. If the EdDSA round 2 API returns a different `sessionId` — as the ECDSA code implies is possible (since it overwrites it) — this assertion will throw unexpectedly.

Please add a comment documenting why EdDSA session IDs are stable across rounds (if that's the API contract), or switch to the ECDSA pattern and remove the assertion.

**Review by Marzooqa — DISMISSED** [review]
Overall structure is correct — the 2-round key-gen flow (no round 3) is intentionally different from ECDSA because MPS EdDSA DKG only requires 2 BitGo rounds. `uploadClientKeys` is a faithful copy of the ECDSA version. Three issues below; one is a potential runtime failure on every call.

**No tests added.** The ECDSA `MPCv2SMCUtils` also has no unit tests, but this is still a gap — at minimum `keyGenRound1BySender` and `keyGenRound2BySender` should have state-assertion failure tests and a happy-path test using a stub sender.

File: modules/sdk-core/src/bitgo/utils/tss/eddsa/SMC/utils.ts [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    });
+
+    const response = {
+      state: EddsaKeyCreationMPCv2StateEnum.WaitingForOVC1Round2Data,
+      tssVersion: payload.tssVersion,
+      walletType: payload.walletType,
+      coin: payload.coin,
+      ovc: payload.ovc,
+      platform: {
+        walletGpgPubKeySigs: (result as EddsaMPCv2KeyGenRound1Response & { walletGpgPubKeySigs: string })
```
Comment (Marzooqa) [inline modules/sdk-core/src/bitgo/utils/tss/eddsa/SMC/utils.ts]: **Type gap on `walletGpgPubKeySigs`.** `EddsaMPCv2KeyGenRound1Response` is typed as `{ sessionId, bitgoMsg1 }` only in `@bitgo/public-types` — `walletGpgPubKeySigs` is absent from the io-ts codec. ECDSA's `MPCv2KeyGenRound1Response` declares it as a required `NonEmptyString` and accesses it directly (`result.walletGpgPubKeySigs`, no cast needed).

If the EdDSA round-1 endpoint returns this field, add it to `EddsaMPCv2KeyGenRound1Response` in `@bitgo/public-types` and drop the cast. If it doesn't, `walletGpgPubKeySigs` will be `undefined` at runtime — and if `EddsaBitgoToOVC1Round1Response.platform.walletGpgPubKeySigs` is a required `NonEmptyString` (as it is in the ECDSA codec), `decodeOrElse` will throw `"error(s) parsing response"` on every call.

File: modules/sdk-core/src/bitgo/utils/tss/eddsa/SMC/utils.ts [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+      source: 'backup',
+      keyType: 'tss',
+      commonKeychain: backupCommonKeychain,
+      isMPCv2: true,
+    });
+
+    const [userKeychain, backupKeychain] = await Promise.all([userKeychainPromise, backupKeychainPromise]);
+    return { userKeychain, backupKeychain, bitgoKeychain };
+  }
+}
```
Comment (Marzooqa) [inline modules/sdk-core/src/bitgo/utils/tss/eddsa/SMC/utils.ts:163]: No unit tests added. At minimum: (1) a state-guard failure test for each of the two `assert` guards (`payload.state !== WaitingForBitgoRound1Data` and `!== WaitingForBitgoRound2Data`), and (2) a happy-path test for `keyGenRound1BySender` and `keyGenRound2BySender` using a stub `senderFn` that returns a mock API response. ECDSA `MPCv2SMCUtils` also lacks tests, but that's not a reason to skip them here. Follow-up safe.

**Review by Marzooqa — COMMENTED** [review]
See inline comments.

File: modules/sdk-core/src/bitgo/utils/tss/eddsa/SMC/utils.ts (hunk 4 above)
Comment (kisslove-dewangan) [reply modules/sdk-core/src/bitgo/utils/tss/eddsa/SMC/utils.ts:163]: @claude can you please create a linear ticket under WCI for this test case task as mentioned above. assigned the ticket to me

File: modules/sdk-core/src/bitgo/utils/tss/eddsa/SMC/utils.ts (hunk 1 above)
Comment (kisslove-dewangan) [reply modules/sdk-core/src/bitgo/utils/tss/eddsa/SMC/utils.ts:65]: resolved, removed unnecessary field

File: modules/sdk-core/src/bitgo/utils/tss/eddsa/SMC/utils.ts (hunk 2 above)
Comment (kisslove-dewangan) [reply modules/sdk-core/src/bitgo/utils/tss/eddsa/SMC/utils.ts:98]: added comment

File: modules/sdk-core/src/bitgo/utils/tss/eddsa/SMC/utils.ts (hunk 3 above)
Comment (kisslove-dewangan) [reply modules/sdk-core/src/bitgo/utils/tss/eddsa/SMC/utils.ts]: resolved

File: modules/sdk-core/src/bitgo/utils/tss/eddsa/SMC/utils.ts (hunk 4 above)
Comment (kisslove-dewangan) [reply modules/sdk-core/src/bitgo/utils/tss/eddsa/SMC/utils.ts:163]: added

**Comment by lokesh-bitgo** [comment]
@claude Please do a thorough review of this PR.

Focus especially on:

- any bugs or incorrect logic in the changes
- whether these changes could impact existing flows or cause regressions
- whether the implementation is aligned with the requirement/design
- whether the changes are correct and complete
- missing edge cases, validation gaps, and test coverage gaps

Please call out concrete issues clearly and separate them by severity if possible.

**Comment by lokesh-bitgo** [comment]
@claude

**Comment by kisslove-dewangan** [comment]
note: API spec breaking is intentional as I've confirmed from change owner, thread: https://bitgo.slack.com/archives/C057BHBRG4B/p1779958910469849

**Review by lokesh-bitgo — APPROVED** [review]
Overall looks good. Please add happy path test for the keyGenRound2 (enterprise) enterprise wrapper in follow up PR.
